crashtracker-ffi/src/collector_windows/api.rs

```diff
@@ -0,0 +1,737 @@
+    let thread_handle = OpenThread(THREAD_ALL_ACCESS, false, thread_id)?;
+    let mut context = AlignedContext::default();
+
+    #[cfg(target_arch = "x86_64")]
+    {
+        context.ctx.ContextFlags = CONTEXT_FULL_AMD64;
+    }
+    #[cfg(target_arch = "x86")]
+    {
+        context.ctx.ContextFlags = CONTEXT_FULL_X86;
+    }
+
+    GetThreadContext(thread_handle, &mut context.ctx)?;
+
+    let mut native_frame = STACKFRAME_EX::default();
+    let machine_type: u32;
+
+    #[cfg(target_arch = "x86_64")]
+    {
+        machine_type = IMAGE_FILE_MACHINE_AMD64.0 as u32;
+        native_frame.AddrPC.Offset = context.ctx.Rip;
+        native_frame.AddrPC.Mode = AddrModeFlat;
+        native_frame.AddrStack.Offset = context.ctx.Rsp;
+        native_frame.AddrStack.Mode = AddrModeFlat;
+        native_frame.AddrFrame.Offset = context.ctx.Rbp;
+        native_frame.AddrFrame.Mode = AddrModeFlat;
+    }
+
+    #[cfg(target_arch = "x86")]
+    {
+        machine_type = IMAGE_FILE_MACHINE_I386.0 as u32;
+        native_frame.AddrPC.Offset = context.ctx.Eip as u64;
+        native_frame.AddrPC.Mode = AddrModeFlat;
+        native_frame.AddrStack.Offset = context.ctx.Esp as u64;
+        native_frame.AddrStack.Mode = AddrModeFlat;
+        native_frame.AddrFrame.Offset = context.ctx.Ebp as u64;
+        native_frame.AddrFrame.Mode = AddrModeFlat;
+    }
+
+    loop {
+        let result = StackWalkEx(
+            machine_type,
+            process_handle,
+            thread_handle,
+            &mut native_frame,
+            &mut context as *mut _ as *mut c_void,
+            None,
+            None,
+            None,
+            None,
+            SYM_STKWALK_DEFAULT,
+        );
+
+        if result != TRUE {
+            break;
+        }
+
+        let mut frame = StackFrame::new();
+
+        frame.ip = Some(format!("{:x}", native_frame.AddrPC.Offset));
+        frame.sp = Some(format!("{:x}", native_frame.AddrStack.Offset));
+
+        // Find the module
+        let module = modules.iter().find(|module| {
+            module.base_address <= native_frame.AddrPC.Offset
+                && native_frame.AddrPC.Offset < module.base_address + module.size
+        });
+
+        if let Some(module) = module {
+            frame.module_base_address = Some(format!("{:x}", module.base_address));
+            frame.symbol_address = Some(format!(
+                "{:x}",
+                native_frame.AddrPC.Offset - module.base_address
+            ));
+            frame.path.clone_from(&module.path);
+
+            if let Some(pdb_info) = &module.pdb_info {
+                frame.build_id = Some(format!("{:x}{:x}", pdb_info.signature, pdb_info.age));
+                frame.build_id_type = Some(datadog_crashtracker::BuildIdType::PDB);
+                frame.file_type = Some(datadog_crashtracker::FileType::PE);
+            }
+        }
+
+        stacktrace
+            .push_frame(frame, true)
+            .expect("Failed to add frame");
```

Comment:
```suggestion
            .context("Failed to add frame")?;
```

Reply:
Done in https://github.com/DataDog/libdatadog/pull/892/commits/ea8e7c31f38e2b3c86a21406ac52785ae29e7797